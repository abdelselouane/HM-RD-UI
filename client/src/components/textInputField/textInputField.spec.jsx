/* global describe, beforeAll, it, expect, jest */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TextInputField from './index';

describe('<TextInputField />', () => {
  let wrapper;
  let props;

  beforeAll(() => {
    React.createRef = jest.fn(() => ({ current: { focus: jest.fn() } }));

    props = {
      className: 'mock-class-name',
      id: 'mockTextInputField',
      onChange: jest.fn(),
      setFocusOnMount: false
    };

    wrapper = shallow(<TextInputField {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({
      icon: {
        className: 'mock_icon'
      }
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('it should call focus on inputRef.current if setFocusOnMount is true', () => {
    wrapper.setProps({
      setFocusOnMount: true
    });
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance.inputRef.current, 'focus');
    instance.componentDidMount();
    expect(spy).toHaveBeenCalled();
  });

  it('should call onChange function on input change', () => {
    const spy = jest.spyOn(props, 'onChange');
    wrapper
      .find('input.mock-class-name')
      .simulate('change', { target: { value: '123' } });

    expect(spy).toHaveBeenCalledWith('mockTextInputField', '123');
    spy.mockReset();
    spy.mockRestore();
  });
});
