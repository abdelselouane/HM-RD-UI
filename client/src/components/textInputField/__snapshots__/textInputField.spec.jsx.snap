// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInputField /> should render properly 1`] = `
<div
  className="text-input-container mock-class-name"
>
  <input
    className="mock-class-name"
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`<TextInputField /> should render properly 2`] = `
<div
  className="text-input-container mock-class-name"
>
  <input
    className="mock-class-name"
    onChange={[Function]}
    type="text"
  />
  <i
    className="mock_icon"
  />
</div>
`;
